analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -358,6 +358,7 @@ private DayActivity createNewDayActivity(UserAnonymized userAnonymizedEntity, Ac
 	{
 		DayActivity dayActivity = DayActivity.createInstance(userAnonymizedEntity, matchingGoal, payload.startTime.getZone(),
 				TimeUtil.getStartOfDay(payload.userAnonymized.getTimeZone(), payload.startTime).toLocalDate());
+		dayActivityRepository.save(dayActivity);
```

Comment:
Indeed, and most of the times it works, but sometimes not. You had that error last week and I had it this week. Explicitly saving it is not an issue and we do this often, but it shouldn't be necessary.

Sometime back, I tried removing all saves except for the root of the cascading tree (see [YD-457](http://jira.yona.nu/browse/YD-457), but that failed because at times, JPA/Hibernate seems to miss a cascading persist.